gateway/src/shard/impl.rs

```diff
@@ -260,4 +261,22 @@ impl Shard {
         self.0.listeners.remove_all().await;
         session.stop_heartbeater().await;
     }
+
+    /// This will shut down the shard in a resumable way and return the info required to resume later
+    pub async fn shutdown_resumable(&self) -> (u64, Option<String>, u64) {
```

Comment:
It's not clear what the tuple values are, maybe this should be a struct instead?

Reply:
lol, when i brought this up in another place i was told it was prob not worth the effort cause you can find it from the context, but sure